chain-abci/src/enclave_bridge/edp/mod.rs

```diff
@@ -10,124 +11,41 @@ use enclave_runner::{
 };
 use parity_scale_codec::{Decode, Encode};
 use sgxs_loaders::isgx::Device;
-use std::io::{Cursor, Read};
-use std::sync::{mpsc::channel, mpsc::Receiver, mpsc::Sender, Arc, Mutex};
+use std::io::{Read, Write};
+use std::sync::{mpsc::channel, Arc, Mutex};
 use std::thread::{self};
-use std::{
-    future::Future,
-    io,
-    pin::Pin,
-    task::{Context, Poll},
-};
-use tokio::io::{AsyncRead, AsyncWrite};
+use std::{future::Future, io, pin::Pin};
 
 use ra_sp_server::{config::SpRaConfig, server::SpRaServer};
 use tokio::net::{TcpListener, TcpStream};
 
 use enclave_utils::zmq_helper::ZmqHelper;
+use std::os::unix::net::UnixStream;
```

Comment:
Shouldn't this be [`tokio::net::UnixStream`](https://docs.rs/tokio/0.2.22/tokio/net/struct.UnixStream.html)?

Reply:
no, this is created in non-Tokio context. For tokio, it's wrapped in the future: https://github.com/crypto-com/chain/pull/2105/files#diff-66e1762a50f6eae106107d439fca4e56R86

Reply:
`tokio::net::UnixStream::from_std` will panic if the runtime isn't set: https://docs.rs/tokio/0.2.22/tokio/net/struct.UnixStream.html#method.from_std

Reply:
what's the benefit of using unix socket?
previous code also looking fine

Reply:
it's >50% less code :) also, before even if there was no request, it still returned "Ready", which could perhaps cause the busy-waiting you mentioned -- now, sleeping/waking should be handled